src/middleware/routeHandler.js

```diff
@@ -40,32 +55,72 @@ const attachRollbackRouteHandlerWrapper = (routeHandler) => async (
   const { siteName } = req.params
 
   const { accessToken } = userSessionData
+  const isRepoWhitelisted = WHITELISTED_GIT_SERVICE_REPOS.split(",").includes(
+    siteName
+  )
 
   await lock(siteName)
 
   let originalCommitSha
-  try {
-    const { currentCommitSha, treeSha } = await getCommitAndTreeSha(
+  if (isRepoWhitelisted) {
+    const result = await gitFileSystemService.getLatestCommitOfBranch(
       siteName,
-      accessToken
+      BRANCH_REF
     )
+    if (result.isErr()) {
+      await unlock(siteName)
+      next(result.err)
+    } else {
+      originalCommitSha = result.value.sha
+      if (!originalCommitSha) {
```

Comment:
when would this occur?

Reply:
This is mostly just a guard clause, it shouldn't occur normally!

Reply:
okay!